titus-server-master/src/test/java/com/netflix/titus/master/integration/v3/scenario/ScenarioTemplates.java

```diff
@@ -57,46 +62,42 @@
         return jobScenarioBuilder -> {
             jobScenarioBuilder.template(startJob(TaskState.Started));
-
-            if (lock) {
-                jobScenarioBuilder.allTasks(taskScenarioBuilder -> {
-                    taskScenarioBuilder.getTaskExecutionHolder().delayStateTransition(state -> Long.MAX_VALUE);
-                    return taskScenarioBuilder;
-                });
-                return jobScenarioBuilder.allTasks(taskScenarioBuilder -> taskScenarioBuilder
-                        .killTask()
-                        .expectStateUpdates(TaskState.KillInitiated)
-                );
-            }
-
-            return jobScenarioBuilder.allTasks(taskScenarioBuilder -> taskScenarioBuilder
-                    .killTask()
-                    .expectStateUpdates(TaskState.KillInitiated, TaskState.Finished)
-            );
+            jobScenarioBuilder.getKube().allowPodTermination(false);
+            return jobScenarioBuilder.allTasks(TaskScenarioBuilder::moveToKillInitiated);
         };
     }
 
     public static Function<JobScenarioBuilder, JobScenarioBuilder> startTasks() {
-        return jobScenarioBuilder -> jobScenarioBuilder
-                .allTasks(taskScenarioBuilder -> taskScenarioBuilder.expectStateUpdateSkipOther(TaskState.Launched))
-                .allTasks(taskScenarioBuilder -> taskScenarioBuilder.transitionTo(TaskState.StartInitiated, TaskState.Started))
-                .allTasks(taskScenarioBuilder -> taskScenarioBuilder.expectStateUpdates(TaskState.StartInitiated, TaskState.Started));
-    }
-
-    public static Function<JobScenarioBuilder, JobScenarioBuilder> startLaunchedTasks() {
-        return jobScenarioBuilder -> jobScenarioBuilder
-                .allTasks(taskScenarioBuilder -> taskScenarioBuilder.transitionTo(TaskState.StartInitiated, TaskState.Started))
-                .allTasks(taskScenarioBuilder -> taskScenarioBuilder.expectStateUpdates(TaskState.StartInitiated, TaskState.Started));
+        return jobScenarioBuilder -> {
+            if (jobScenarioBuilder.isKube()) {
+                return jobScenarioBuilder
+                        .schedule()
+                        .allTasks(taskScenarioBuilder -> taskScenarioBuilder.expectStateUpdates(TaskState.Launched))
+                        .allTasks(taskScenarioBuilder -> taskScenarioBuilder.transitionTo(TaskState.StartInitiated, TaskState.Started))
+                        .allTasks(taskScenarioBuilder -> taskScenarioBuilder.expectStateUpdates(TaskState.StartInitiated, TaskState.Started));
+            }
+            return jobScenarioBuilder
+                    .allTasks(taskScenarioBuilder -> taskScenarioBuilder.expectStateUpdateSkipOther(TaskState.Launched))
+                    .allTasks(taskScenarioBuilder -> taskScenarioBuilder.transitionTo(TaskState.StartInitiated, TaskState.Started))
+                    .allTasks(taskScenarioBuilder -> taskScenarioBuilder.expectStateUpdates(TaskState.StartInitiated, TaskState.Started));
```

Comment:
do we still need this code path? Shouldn't everything be `isKube() => true`?

Reply:
if we do need this, perhaps line 68 also needs a `isKube()` check?

Reply:
There is a lot of cleanup required to remove Fenzo code branches (+ Mesos CloudSimulator). I am going to do it in the next iteration after merging this PR.